src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -437,18 +445,33 @@ def find_candidates(
             and all(req.is_satisfied_by(c) for req in requirements[identifier])
         )
 
-    def _make_requirement_from_install_req(
+    def _make_requirements_from_install_req(
         self, ireq: InstallRequirement, requested_extras: Iterable[str]
-    ) -> Optional[Requirement]:
+    ) -> List[Requirement]:
```

Comment:
I wonder if this should be an iterator (generator) instead, with the `list` call moved to the call site if needed (since some sites don’t need this to be a list).

Reply:
Yes, good idea. Applied in 21bfe40.

Reply:
Forgot to push and linked the wrong commit: applied in 0de374e4d for real.